### PR TITLE
feat: add compliance tags to `kili.update_properties_in_project`

### PR DESCRIPTION
**Pull request by Jonas1312**

### CODE REVIEW

File: tests/integration/use_cases/test_project.py [hunk 1]
```diff
@@ -57,3 +59,47 @@ def test_when_i_query_projects_i_get_a_generator_of_projects(kili_api_gateway: K
...
+    def mocked_update_properties_in_project(
+        project_id: ProjectId,
+        project_data: ProjectDataKiliAPIGatewayInput,
+        fields: ListOrTuple[str],
+    ):
+        return {field: f"{field}_value" for field in fields}
+
+    kili_api_gateway.update_properties_in_project.side_effect = mocked_update_properties_in_project
+
+    # Given
```
Comment (PierreLeveau) [inline tests/integration/use_cases/test_project.py:76]: move up

File: tests/integration/presentation/test_project.py [hunk 2]
```diff
@@ -28,3 +31,46 @@ def test_when_creating_project_then_it_returns_project_id(mocker: pytest_mock.Mo
             }
         },
     )
+
+
+def test_when_updating_project_then_it_returns_updated_project(mocker: pytest_mock.MockerFixture):
+    kili = ProjectClientMethods()
```
Comment (PierreLeveau) [inline tests/integration/presentation/test_project.py:37]: `# Given`

**Review by PierreLeveau — APPROVED** [review]

**Review by PierreLeveau — APPROVED** [review]

File: src/kili/adapters/kili_api_gateway/project/__init__.py [hunk 3]
```diff
@@ -79,3 +84,17 @@ def list_projects(
...
+
+    def update_properties_in_project(
+        self,
+        project_id: ProjectId,
+        project_data: ProjectDataKiliAPIGatewayInput,
+        fields: ListOrTuple[str],
+    ) -> Dict:
+        """Update properties in a project."""
+        fragment = fragment_builder(fields)
+        query = get_update_properties_in_project_mutation(fragment)
```
Comment (theodu) [inline src/kili/adapters/kili_api_gateway/project/__init__.py:96]:
```suggestion
        mutation = get_update_properties_in_project_mutation(fragment)
```

File: src/kili/adapters/kili_api_gateway/project/__init__.py [hunk 4]
```diff
@@ -79,3 +84,17 @@ def list_projects(
...
+        data = project_data_mapper(data=project_data)
+        variables = {"data": data, "where": {"id": project_id}}
+        result = self.graphql_client.execute(query, variables)
```
Comment (theodu) [inline src/kili/adapters/kili_api_gateway/project/__init__.py:99]:
```suggestion
        result = self.graphql_client.execute(mutation, variables)
```

File: src/kili/adapters/kili_api_gateway/project/types.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+    def __post_init__(self) -> None:
+        _check_argument_ranges(
+            consensus_tot_coverage=self.consensus_tot_coverage,
+            min_consensus_size=self.min_consensus_size,
+            review_coverage=self.review_coverage,
+        )
+
+
+def _check_argument_ranges(
```
Comment (theodu) [inline src/kili/adapters/kili_api_gateway/project/types.py:46]: it should rather be done in the presentation or in the use case. It is linked to what the user inputs

Comment (theodu) [reply src/kili/adapters/kili_api_gateway/project/types.py:46]: I find it weird to handle this here when the role of the gateway should just be to launch a call to Kili API, not to have logic. I would put it in the presentation layer. I asked to GPT and also answered that checks should be in the presentation for separation of concern. I would really avoid to add anything unnecessary in the gateway whose role should only be to send a call to Kili and to adapt the result to the SDK.

**Pull request merged**